src/webots/gui/WbNewProtoWizard.cpp

```diff
@@ -97,151 +98,185 @@ bool WbNewProtoWizard::validateCurrentPage() {
-              if (!externPath.contains(declaration))
-                externPath += declaration;
+            } else {
+              // handle case where the sub-proto is relative and the PROTO of the base type is local (webots://)
+              if (WbUrl::isLocalUrl(parentUrl) && (!WbUrl::isWeb(nestedUrl) && QDir::isRelativePath(nestedUrl)))
+                nestedUrl = WbUrl::combinePaths(nestedUrl, parentUrl);
             }
+            const QString declaration =
+              QString("EXTERNPROTO \"%1\"\n").arg(nestedUrl.replace(WbStandardPaths::webotsHomePath(), "webots://"));
+            if (!externPath.contains(declaration))
+              externPath += declaration;
           }
         }
+      }
 
-        // define IS connections in the body
-        body += "  " + mBaseNode + " {\n";
-        for (int i = 0; i < parameterNames.size(); ++i)
-          if (mExposedFieldCheckBoxes[i + 1]->isChecked())
-            body += "    " + parameterNames[i] + " IS " + parameterNames[i] + "\n";
-        body += "  }";
-      } else {
-        // define interface
-        const WbNodeModel *const nodeModel = WbNodeModel::findModel(mBaseNode);
-        const QList<WbFieldModel *> fieldModels = nodeModel->fieldModels();
-        for (int i = 0; i < fieldModels.size(); ++i) {
-          if (mExposedFieldCheckBoxes[i + 1]->isChecked()) {
-            const WbValue *defaultValue = fieldModels[i]->defaultValue();
-            QString vrmlDefaultValue = defaultValue->toString();
-
-            if (defaultValue->type() == WB_SF_NODE && vrmlDefaultValue != "NULL")
-              vrmlDefaultValue += "{}";
-
-            const WbMultipleValue *mv = dynamic_cast<const WbMultipleValue *>(defaultValue);
-            if (defaultValue->type() == WB_MF_NODE && mv) {
-              vrmlDefaultValue = "[ ";
-              for (int j = 0; j < mv->size(); ++j)
-                vrmlDefaultValue += mv->itemToString(j) + "{} ";
-              vrmlDefaultValue += "]";
-            }
-            interface +=
-              "  field " + defaultValue->vrmlTypeName() + " " + fieldModels[i]->name() + " " + vrmlDefaultValue + "\n";
+      // if the interface refers to a relative url, turn it into a web one
+      const QRegularExpression resources("^\"([^\"]*)\\.(jpe?g|png|hdr|obj|stl|dae|wav|mp3)$\"",
+                                         QRegularExpression::CaseInsensitiveOption);
+      QRegularExpressionMatchIterator it = resources.globalMatch(interface);
+      while (it.hasNext()) {
+        const QRegularExpressionMatch match = it.next();
+        if (match.hasMatch()) {
+          QString asset = match.captured(0);
+          asset.replace("\"", "");
+          if (!WbUrl::isWeb(asset) && QDir::isRelativePath(asset))
+            interface.replace(QString("\"%1\"").arg(asset), QString("\"%1\"").arg(WbUrl::combinePaths(asset, info->url())));
+        }
+      }
```

Comment:
note: the diff is big but it's because I've removed most of the lines from the body of a if by testing `if (!success)` rather than `if (success)`. L205-218 is the only actual addition done